third-party/realdds/src/dds-device-server.cpp

```diff
@@ -237,114 +256,140 @@ void dds_device_server::on_control_message_received()
+    std::string owner_name;  // default is empty, for a device option
+    rsutils::json::get_ex( j, owner_name_key, &owner_name );
 
     std::shared_ptr< dds_option > opt = find_option( option_name, owner_name );
-
     if( opt )
     {
-        try
-        {
-            float value = rsutils::json::get< float >( j, "value" );
+        float value = rsutils::json::get< float >( j, value_key );
+        if( _set_option_callback )
             _set_option_callback( opt, value ); //Handle setting option outside realdds
-            opt->set_value( value ); //Update option object. Do second to check if _set_option_callback did not throw
-            send_set_option_success( counter );
-        }
-        catch( std::exception e )
-        {
-            send_set_option_failure( counter, e.what() );
-        }
+        opt->set_value( value ); //Update option object. Do second to check if _set_option_callback did not throw
+        reply[value_key] = value;
     }
     else
     {
-        send_set_option_failure( counter, owner_name + " does not support option " + option_name );
+        if( owner_name.empty() )
+            owner_name = "device";
+        else
+            owner_name = "'" + owner_name + "'";
+        DDS_THROW( runtime_error, owner_name + " option '" + option_name + "' not found" );
     }
 }
 
-void dds_device_server::handle_query_option( const nlohmann::json & j )
+
+void dds_device_server::handle_query_option( const nlohmann::json & j, nlohmann::json & reply )
 {
-    const std::string & option_name = rsutils::json::get< std::string >( j, "option-name" );
-    const std::string & owner_name = rsutils::json::get< std::string >( j, "owner-name" );
-    uint32_t counter = rsutils::json::get< uint32_t >( j, "counter" );
+    auto option_name = rsutils::json::get< std::string >( j, option_name_key );
+    std::string owner_name;  // default is empty, for a device option
+    rsutils::json::get_ex( j, owner_name_key, &owner_name );
 
     std::shared_ptr< dds_option > opt = find_option( option_name, owner_name );
-
     if( opt )
     {
-        try
+        float value;
+        if( _query_option_callback )
         {
-            float value = _query_option_callback( opt ); //Handle query outside realdds
-            opt->set_value( value ); //Ensure realdds option is up to date with actual value (might change, e.g temperature)
-            send_query_option_success( counter, value );
+            value = _query_option_callback( opt );
+            // Ensure realdds option is up to date with actual value from callback
+            opt->set_value( value );
         }
-        catch( std::exception e )
+        else
         {
-            send_query_option_failure( counter, e.what() );
+            value = opt->get_value();
         }
+        reply[value_key] = value;
     }
     else
     {
-        send_query_option_failure( counter, owner_name + " does not support option " + option_name );
+        if( owner_name.empty() )
```

Comment:
You can also initialize `owner_name` to "Device" and omit the ticks around stream name, than delete this `if` statement

Reply:
I wanted the ticks around the stream name, and not for the `device` which is not a name.